Application/Frontend/components/Messaging/MessageDropdown.tsx

```diff
@@ -0,0 +1,80 @@
+import { Menu, Button, Text, rem, ActionIcon, useComputedColorScheme, Tooltip } from '@mantine/core';
+import {
+  IconTrash,
+  IconDotsVertical,
+  IconCursorText
+} from '@tabler/icons-react';
+import { MessageDropdownProps } from '@/accordTypes';
+import { useState, useEffect, forwardRef, ReactNode } from 'react';
+import { useUser } from '@clerk/nextjs';
+
+export function MessageDropdown({ privateChat, clientID, onDelete }: MessageDropdownProps) {
+  const { user } = useUser();
+  const [userID, setUserID] = useState<string | null>(null);
+  const [readyToDelete, setReadyToDelete] = useState(false);
+  const textColor = useComputedColorScheme() === 'dark' ? "white" : "black";
+
+  // Effect to check for user id availability
+  useEffect(() => {
+    if (user && user.id) {
+      setUserID(user.id);
+      setReadyToDelete(true);
+    }
+  }, [user?.id]);
+
+  // Check if it's the current user's message
+  const myMessage = clientID === userID;
+
+  const MenuItemWithOptionalTooltip = forwardRef<HTMLDivElement, { children: ReactNode, privateChat: boolean, onDelete: () => void }>(({ children, privateChat, onDelete }, ref) => {
```

Comment:
- You've attached the onDelete handler to the onClick event of the Menu.Item. However, since you're already passing onDelete to MenuItemWithOptionalTooltip, it might be redundant to attach it again directly to Menu.Item. You can consider removing it from Menu.Item if it's already handled within MenuItemWithOptionalTooltip.